### DISCUSSION TITLE
How to disable security algorithms

### DISCUSSION THREAD

**Discussion by vijaywork28** [srsran/srsRAN_Project#795] [General Help]
Hi,
can we disable integrity and ciphering for SRBs ( signalling radio bearers) ?

If it is possible, kindly help us how to disable security algorithms.

Thanks in advance

**Comment by herlesupreeth** [comment]
Please take a look at `cu_cp security --nea_pref_list=nea0,nea2,nea1,nea3 --nia_pref_list=nia2,nia1,nia3` configuration option.

I dont think integrity can be disabled. @robertfalkenberg can help you with it more details

**Reply by robertfalkenberg** [reply]
That's right, SRB integrity cannot be disabled. To disable SRB ciphering, put `nea0` (null ciphering) as first entry in the `nea_pref_list` as listed the example @herlesupreeth posted above.

**Reply by vijaywork28** [reply]
Thank you for the clarification

**Reply by PiscesDR** [reply]
Hi,
could you please tell me where is the configuration entry to enable encryption and integrity protection for SRB? I use ZMQ to implement communication between srsUE and srsGNB. The default is NEA0 and NIA2. I want to change this configuration, but I can't find the content you mentioned.
Thanks a lot.

**Reply by robertfalkenberg** [reply]
add `cu_cp security --confidentiality=required --integrity=required --nea_pref_list=nea2 --nia_pref_list=nia2`
